### DISCUSSION TITLE
Rename git master and Docker :latest

### DISCUSSION THREAD

**Discussion by Alkarex** [FreshRSS/FreshRSS#3296] [Strategy]
Let's give a try to this new *Discussion* area 🤓

I suggest renaming our git *master* branch to *edge*.
Accordingly on Docker Hub, our latest release would be `:latest`(i.e. the default) while the build from git would be `:edge`.

This terminology is already used by popular projects such as https://hub.docker.com/_/alpine/

**NB**: This is a change from our current tagging, which uses `:latest` for the master git build version, but the proposed change would be more standard, and there is a demand https://github.com/FreshRSS/FreshRSS/issues/3324 , https://github.com/FreshRSS/FreshRSS/issues/3109 , https://github.com/FreshRSS/FreshRSS/issues/3039

On GitHub, [others are renaming *master* to *main*](https://github.com/github/renaming) but I find this wording a bit confusing in our case.

Other projects such as https://hub.docker.com/_/debian are using `:latest` (which is the latest stable release), `:testing`, `:unstable`

Thoughts?

@marienfressinaud @aledeg @Frenzie

**Comment by Frenzie** [comment]
I'm not sure what I think of `edge`, but `experimental` would be longer so probably worse. You could also think in the direction of `risky`. ;-) I kind of like Debian's `unstable`, but I should probably add I've been using Debian as my main OS for close to a decade now.

> On GitHub, others are renaming master to main but I find this wording a bit confusing in our case.

I think main and master mean the same thing, or very close to it. But I don't really mind what it's called as long as it's the default (which is another word you might consider, I suppose).

**Comment by Alkarex** [comment]
We also used `:dev`, which could be an option.
In any case, I would like to have the same wording on GitHub and Docker Hub.

**Reply by aledeg** [reply]
Having the same wording on github and docker hub makes sense. But the name itself is not a big deal.
`edge`, `default`, `unstable`, they are all good.

**Comment by marienfressinaud** [comment]
Let's vote to see if an option is more appreciated than the others. We can vote for several suggestions. If there's an equality, we'll use `/dev/random` :)

Have been named for the competition: `edge`, `testing`, `unstable`, `experimental`, `risky`, `dev` and `default`.

Does this sound good to you? Any other suggestion?

**Reply by Frenzie** [reply]
`delicate` :-P

**Comment by Alkarex** [comment]
Done:

## Git

* `edge` = our master / main branch, rolling release
* `latest` https://github.com/FreshRSS/FreshRSS/releases/latest = our latest release

## Docker Hub

* `freshrss/freshrss:latest` = latest release based on Debian
* `freshrss/freshrss:latest-alpine` = latest release based on Alpine Linux
* `freshrss/freshrss:latest-arm` = latest release based on Debian for ARM

* `freshrss/freshrss:edge` = git build rolling release based on Debian
* `freshrss/freshrss:edge-alpine` = git build rolling release based on Alpine Linux
* `freshrss/freshrss:edge-arm` = git build rolling release based on Debian for ARM

I will document as part of https://github.com/FreshRSS/FreshRSS/issues/3351

P.S. I think we should look into using GitHub Actions to push to Docker Hub instead of doing it all from Docker Hub, in order in particular to have a Docker-specific README on https://hub.docker.com/r/freshrss/freshrss/ (other ideas welcome)
